pkg/apis/kubebind/v1alpha1/apiservicebinding_types.go

```diff
@@ -87,8 +87,265 @@ type APIServiceBindingSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="kubeconfigSecretRef is immutable"
 	KubeconfigSecretRef ClusterSecretKeyRef `json:"kubeconfigSecretRef"`
+
+	// permissionClaims records decisions about permission claims requested by the API service provider.
+	// Individual claims can be accepted or rejected. If accepted, the API service provider gets the
+	// requested access to the specified resources in this workspace. Access is granted per
+	// GroupResource and other properties like selectors.
+	//
+	// +optional
+	PermissionClaims []AcceptablePermissionClaim `json:"permissionClaims,omitempty"`
+}
+
+// acceptablePermissionClaim is a permission claim that stores the users acceptance in the field state. Only accepted permission claims are reconciled.
+type AcceptablePermissionClaim struct {
+	PermissionClaim `json:",inline"`
+
+	// state indicates if the claim is accepted or rejected.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Accepted;Rejected
+	State AcceptablePermissionClaimState `json:"state"`
 }
 
+type AcceptablePermissionClaimState string
+
+const (
+	ClaimAccepted AcceptablePermissionClaimState = "Accepted"
+	ClaimRejected AcceptablePermissionClaimState = "Rejected"
+)
+
+// permissionClaim selects objects of a GVR that a service provider may
+// request and that a consumer may accept and allow the service provider access to.
+//
+// +kubebuilder:validation:XValidation:rule="!(has(self.autoDonate) && self.autoDonate && has(self.autoAdopt) && self.autoAdopt)",message="donate and adopt are mutually exclusive"
+type PermissionClaim struct {
+	GroupResource `json:","`
+
+	// version is the version of the claimed resource.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength:=1
+	Version string `json:"version"`
+
+	// selector selects which resources are being claimed.
+	// If unset, all resources across all namespaces are being claimed.
+	//
+	// +optional
+	// +kubebuilder:default:={}
+	Selector *ResourceSelector `json:"selector,omitempty"`
+
+	// required indicates whether the APIServiceBinding will work if this claim is not accepted. If a required claim is denied, the binding is aborted.
+	Required bool `json:"required"`
+
+	// read claims read access to matching objects for the provider.
+	// reading of the claimed object(s) is always claimed.
+	// By default no labels and annotations are read. Reading of labels and annotations can be claimed
+	// optionally by adding labels and annotations items.
+	// If labels on consumer owned objects that are set by the consumer are read, labelsOnProviderOwnedObjects and
+	// annotationsOnProviderOwnedObjects can be set.
```

Comment:
```suggestion
	// read claims read access to matching objects for the provider.
	// Reading of the claimed object(s) is always claimed.
	// By default, no labels and annotations can be read by the provider. 
	// Reading of labels and annotations can be claimed in addition by specifying them explicitly.
	// If labels on consumer owned objects that are set by the consumer are read, labelsOnProviderOwnedObjects and
	// annotationsOnProviderOwnedObjects can be set.
```

Still trying to make sense of the last sentence. Might come back here during review.